app/src/test/java/com/codesoom/assignment/controller/ProductControllerTest.java

```diff
@@ -0,0 +1,273 @@
+package com.codesoom.assignment.controller;
+
+import com.codesoom.assignment.domain.Product;
+import com.codesoom.assignment.dto.ProductDto;
+import com.codesoom.assignment.exception.ProductNotFoundException;
+import com.codesoom.assignment.service.ProductService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willThrow;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class ProductControllerTest {
+
+    private final long ID = 1L;
+    private final long NOT_EXIST_ID = 100L;
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @MockBean
+    ProductService productService;
+
+    private List<Product> productList;
+    private Product product;
+    private Product product2;
+
+    @BeforeEach
+    void setUp() {
+        product = new Product(ID, "장난감 뱀", "장난감 컴퍼니", 10000, "뱀.jpg");
+        product2 = new Product(2L, "장난감 곰", "장난감 컴퍼니", 20000, "곰.jpg");
+    }
+
+
+    @Nested
+    @DisplayName("Get /products 요청")
+    class Describe_getProducts {
+        @Nested
+        @DisplayName("서비스에 장난감이 있으면")
```

Comment:
`@BeforeEach`와 손발을 맞춰 보면 어떨까요?
```suggestion
        @DisplayName("서비스에 장난감이 1개 이상 있으면")
```